File: app/src/main/java/com/hellmund/primetime/ui/SplashScreenActivity.java
package com.hellmund.primetime.ui;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatActivity;

import com.hellmund.primetime.R;
import com.hellmund.primetime.ui.introduction.IntroductionActivity;
import com.hellmund.primetime.ui.main.MainActivity;

public class SplashScreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        if (isFirstLaunch()) {
            openIntroduction();
        } else {
            openMain();
        }
    }

    private void openIntroduction() {
        Intent intent = new Intent(this, IntroductionActivity.class);
        startActivity(intent);
        finish();
    }

    private void openMain() {
        Intent intent = new Intent(this, MainActivity.class);

        if (getIntent().getExtras() != null) {
            final String shortcut = (String) getIntent().getExtras().get("intent");
            intent.putExtra("intent", shortcut);
        }

        startActivity(intent);
        finish();
    }

    private boolean isFirstLaunch() {
        SharedPreferences sharedPrefs = PreferenceManager.getDefaultSharedPreferences(this);
        return sharedPrefs.getBoolean("firstLaunchOfPrimeTime", true);
    }

}

File: app/src/main/java/com/hellmund/primetime/splash/SplashScreenActivity.kt
package com.hellmund.primetime.splash

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.hellmund.primetime.R
import com.hellmund.primetime.introduction.IntroductionActivity
import com.hellmund.primetime.main.MainActivity
import com.hellmund.primetime.utils.OnboardingHelper

class SplashScreenActivity : AppCompatActivity() {

    private val onboardingHelper: OnboardingHelper by lazy {
        OnboardingHelper(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        if (true) { // TODO onboardingHelper.isFirstLaunch) {
            openIntroduction()
        } else {
            openMain()
        }
    }

    private fun openIntroduction() {
        val intent = IntroductionActivity.newIntent(this)
        startActivity(intent)
        finish()
    }

    private fun openMain() {
        val intent = MainActivity.newIntent(this, intent.getStringExtra("intent"))
        startActivity(intent)
        finish()
    }

}
